ProcessLib/UncoupledProcessesTimeLoop.cpp

```diff
@@ -916,24 +961,26 @@ bool UncoupledProcessesTimeLoop::solveCoupledEquationSystemsByStaggeredScheme(
             }
 
             // Check the convergence of the coupling iteration
-            auto& x_old = *_solutions_of_last_cpl_iteration[pcs_idx];
-            MathLib::LinAlg::axpy(x_old, -1.0, x);
-            _global_coupling_conv_crit->checkResidual(x_old);
-            coupling_iteration_converged =
-                coupling_iteration_converged &&
-                _global_coupling_conv_crit->isSatisfied();
-
+            auto& x_old = *_solutions_of_last_cpl_iteration[process_id];
+            if (global_coupling_iteration > 0)
+            {
+                MathLib::LinAlg::axpy(x_old, -1.0, x);  // save dx to x_old
+                _global_coupling_conv_crit[process_id]->checkDeltaX(x_old, x);
+                coupling_iteration_converged =
+                    coupling_iteration_converged &&
+                    _global_coupling_conv_crit[process_id]->isSatisfied();
+            }
             MathLib::LinAlg::copy(x, x_old);
 
-            if (coupling_iteration_converged)
+            if (coupling_iteration_converged && global_coupling_iteration > 0)
```

Comment:
Maybe it's better to remove this break statement and the `&& global_coupling_iteration > 0` conditions. Let's discuss.

Reply:
Done as what discussed.